app/services/categories/find_by_nomis_booking_id.rb

```diff
@@ -0,0 +1,19 @@
+module Categories
+  class FindByNomisBookingId
+    attr_reader :nomis_booking_id
+
+    def initialize(nomis_booking_id)
+      @nomis_booking_id = nomis_booking_id
+    end
+
+    def call
+      Category.find_by(key: booking_details[:category_code]) if nomis_booking_id.present?
```

Comment:
@martyn-w is there is a possibility that `booking_details` could be nil here? If so might be worth adding a spec in a follow up PR just to explore the behaviour for that scenario and potentially add a nil guard if needed. I guess for now this will just return a 500 error which might not be the friendliest behaviour for the front end 🤔

Reply:
Hi @smoothcontract I don't see how that could happen?: the `booking_details` variable will either be the an object like this: 
`{category: 'foo', category_code: 'F'}` 
or, if not found, an object like this:
`{category: nil, category_code: nil}` 

see the `app/lib/nomis_client/booking_details.rb` library

Reply:
Im not sure that could happen if we get a successful response from NOMIS, as we are wrapping it within our own hash and keys: https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/app/lib/nomis_client/booking_details.rb#L27
If NOMIS itself does raise an error, we now rescue it in the API controller and return an appropriate message https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/app/controllers/api_controller.rb#L32